### ISSUE TITLE
How to hide all artisan commands except the ones created with this tool

### ISSUE THREAD

**Issue by mfour-merickson** [laravel-zero/laravel-zero#101]
I am looking for a way to hide all the commands (see screenshot, those outlined with red) and only show the commands I have created for our CLI (again, see screenshot, those outlined in green)

![image](https://user-images.githubusercontent.com/31225533/34192742-ca390ede-e504-11e7-93ea-3ff6c5585403.png)

**Comment by eddiriarte** [comment]
just set: `'production' => false,` in `config/app.php`

**Comment by nunomaduro** [comment]
@mfour-merickson Check if the @eddiriarte solution works for you.

**Comment by mfour-merickson** [comment]
@nunomaduro No, that does not help (I have already done that part) Either it does not work (as documented) or it doesn't take into consideration the installation of the database module `install:database`

**Comment by mfour-merickson** [comment]
@eddiriarte I have already done that, and while it removes some items, it does not remove the database stuff.  I knew I should have posted this as well (would have wasted this back and forth)

![image](https://user-images.githubusercontent.com/31225533/34216395-05196654-e55d-11e7-87ab-857a3a41a3c2.png)

**Comment by nunomaduro** [comment]
@mfour-merickson I will check it tonight.

**Comment by eddiriarte** [comment]
they seem like some eloquent commands... you could try to set 
```php
    /*
     * If true, migrations commands will be available.
     */
    'with-migrations' => false,

    /*
     * If true, seeds commands will be available.
     */
    'with-seeds' => false,
```
in your config.... 

i found that these commands are been registered here: `LaravelZero\Framework\Commands\Component\Illuminate\Database\ComponentProvider` 

```php
    /**
     * {@inheritdoc}
     */
    public function register(): void
    {
        if (file_exists(config_path('database.php'))) {
            $this->mergeConfigFrom(config_path('database.php'), 'database');
        }

        $this->registerDatabaseService();

        $this->registerMigrationService();
    }
```
didnt tried but maybe a check for "production" before `$this->registerDatabaseService() ... ` do the fix

**Comment by nunomaduro** [comment]
@mfour-merickson You must set the following configuration on `database.php`:

```
    /*
     * If true, migrations commands will be available.
     */
    'with-migrations' => false,

    /*
     * If true, seeds commands will be available.
     */
    'with-seeds' => false,
```

**Comment by mikeerickson** [comment]
OK, this sort of gives me what I want, however, not quite.

I have added these flags to `database.php` and now they dont appear in the menu

However, my custom commands actually call these artisan commands under the hood.  Thus, apply these changes makes the commands unavailable.   

What I am trying to accomplish is to create a CLI which custom migrate and seed commands, and prevent the user from actually performing these actions in a destructive way.

So, I guess my only alternate solution will be to create a CLI layer (written in node or python) on top of the artisan commands (unless you have a possible workaround).

In short, I don't want the user to be able to call these routines themselves, BUT I still need to call migrate (and db:seed) from within the custom CLI layer

**Comment by nunomaduro** [comment]
@mikeerickson I see multiple solutions to your problem only using Laravel Zero. But first, the project you are working is open sourced?

**Comment by mfour-merickson** [comment]
Not really open sourced, it is an internal admin CLI (to manage some docker databases).  We use laravel migrations as our source of truth for our databases.


Mike Erickson
Engineering Manager
MFour Mobile Research
(714) 754-1234 <tel:(714)%20754-1234>

MFour Blog <https://mfour.com/blog/>

**Comment by nunomaduro** [comment]
@mfour-merickson One solution: **Resolve commands on runtime**.

On your `DatabaseBuildCommand.php`:

```
    /**
     * Execute the command. Here goes the code.
     *
     * @return void
     */
    public function handle(): void
    {
        // Adds the core commands that you need on runtime.
        $this->getApplication()->addCommands([
            app(\Illuminate\Database\Console\Migrations\MigrateMakeCommand::class)
        ]);

        // Call them.
        $this->call('make:migration', ['name' => 'creates_issue101_table']);
    }
```

**Comment by nunomaduro** [comment]
@mfour-merickson @mikeerickson There are other solutions, if you don't like the first one.

**Comment by mikeerickson** [comment]
@nunomaduro I will give it a whirl and see how it goes.... If this simple approach works, that is fine by me.

**Comment by mikeerickson** [comment]
@nunomaduro Actually, I don't need to create migrations, I just need to execute migration:refresh and db:seed, but I am sure the same approach will apply.

**Comment by mikeerickson** [comment]
@nunomaduro Which class should be loaded at runtime to allow use of `migrate:refresh`

**Comment by nunomaduro** [comment]
@mikeerickson To have 0 problems:

```
        $this->getApplication()->addCommands(collect([
            \Illuminate\Database\Console\Migrations\FreshCommand::class,
            \Illuminate\Database\Console\Migrations\InstallCommand::class,
            \Illuminate\Database\Console\Migrations\MigrateCommand::class,
            \Illuminate\Database\Console\Migrations\RefreshCommand::class,
            \Illuminate\Database\Console\Migrations\ResetCommand::class,
            \Illuminate\Database\Console\Migrations\RollbackCommand::class,
            \Illuminate\Database\Console\Migrations\StatusCommand::class,
            \Illuminate\Database\Console\Seeds\SeedCommand::class,
        ])->map(function ($command) {
            return app($command);
        })->toArray());

        $this->call('migrate:refresh', ['--force' => true]);
```

**Comment by mikeerickson** [comment]
@nunomaduro  I see the point here, I am just going to go the safe route, but if it becomes "too much" then I can clearly see what needs to be used.   Thanks a bunch, I have a working solution (and I learned something new along the way)

**Comment by nunomaduro** [comment]
No problem! 👊

**Comment by mikeerickson** [comment]
Thanks for your quick response, greatly appreciated.  And thanks for this nifty tool, I am growing to like it more and more.  It will surely come in handy with our future efforts

**Comment by JuanLeadSupply** [comment]
I used the following way of hiding the unwanted options in the Phar file.

I just add to the commands.remove configuration key:

```
 'remove' => Phar::running() ? [
        Illuminate\Database\Console\Migrations\FreshCommand::class,
        Illuminate\Database\Console\Migrations\InstallCommand::class,
        Illuminate\Database\Console\Migrations\MigrateCommand::class,
        Illuminate\Database\Console\Migrations\RefreshCommand::class,
        Illuminate\Database\Console\Migrations\ResetCommand::class,
        Illuminate\Database\Console\Migrations\RollbackCommand::class,
        Illuminate\Database\Console\Migrations\StatusCommand::class,
        Illuminate\Database\Console\Migrations\MigrateMakeCommand::class,
        Illuminate\Database\Console\Seeds\SeedCommand::class,
        Illuminate\Database\Console\WipeCommand::class,
        Illuminate\Database\Console\Factories\FactoryMakeCommand::class,
        Illuminate\Foundation\Console\ModelMakeCommand::class,
        Illuminate\Database\Console\Seeds\SeederMakeCommand::class,
        LaravelZero\Framework\Commands\MakeCommand::class,
        LaravelZero\Framework\Commands\RenameCommand::class,
        LaravelZero\Framework\Commands\StubPublishCommand::class,
        LaravelZero\Framework\Commands\BuildCommand::class,
        LaravelZero\Framework\Commands\InstallCommand::class,
    ] : [],
```
`
